samples/local_attestation/common/attestation.cpp

```diff
@@ -91,25 +94,27 @@ bool Attestation::attest_local_report(
 
     TRACE_ENCLAVE("report_size = %ld", report_size);
 
+    oe_verifier_initialize();
+
     // 1)  Validate the report's trustworthiness
     // Verify the report to ensure its authenticity.
-    result = oe_verify_report(local_report, report_size, &parsed_report);
+    result = oe_verify_evidence(&selected_format, local_report, report_size, NULL, 0, NULL, 0, &claims, &claims_length);
     if (result != OE_OK)
     {
-        TRACE_ENCLAVE("oe_verify_report failed (%s).\n", oe_result_str(result));
+        TRACE_ENCLAVE("oe_verify_evidence failed (%s).\n", oe_result_str(result));
         goto exit;
     }
 
-    TRACE_ENCLAVE("oe_verify_report succeeded\n");
+    TRACE_ENCLAVE("oe_verify_evidence succeeded\n");
 
     // 2) validate the enclave identity's signed_id is the hash of the public
     // signing key that was used to sign an enclave. Check that the enclave was
     // signed by an trusted entity.
-    if (memcmp(parsed_report.identity.signer_id, m_enclave_mrsigner, 32) != 0)
+    if (memcmp(claims[4].value, m_enclave_mrsigner, 32) != 0)
```

Comment:
why `4`? can you add comments (e.g., how claims should be parsed) and consider using `#define` to improve the readability/maintainability of the code.

Besides, is there better way to parse claim rather than treating it as an array and using hard-coded constant to look up?

Reply:
We have had some ongoing conversations about some problems with the new claims format. Essentially I think there are two ways to go about this:

1. Iterate through the claims list and check against the claims name for each possible claim. This method is demonstrated in common/attest_plugin.c. The problem with this method is that it makes the code extremely cumbersome, and in addition, there is outdated documentation wherein not all known claims are returned, such as validity_from and validity_to. The benefit to this is it improves readability.

2. The current way shown. This makes the code more succinct, but like you said, it relies on a prior knowledge of the claims order and it is hard to understand what something like claims[4] means exactly.

My opinion is that the plugin developer should enumerate the claims that they insert into the claims list (using #define or whatever). Then the consumer of the claims list should index into the claims list using those descriptive constants. Since this sample is demonstrating consumption of an existing SGX plugin, I don't think we should include #define here, because it is the plugin author that controls the order in which claims are inserted, so it doesn't make sense for someone just using oe_verify_evidence() to have to dig into code to find out the naming convention/ordering of well-known and custom claims. More generally, all plugins included by default should have documentation on how they handle claims.

Reply:
@yentsanglee What is your opinion on the most ideal way to parse these claims?

Reply:
I prefer option 1 (iterate and check against the claims name).  I don't think we should suggest ever depending on the claims order.

Reply:
This is a good feedback. Option 1 seems cumbersome but it's better option of the 2. I will take this feedback into next API review.

Reply:
@yentsanglee I used option 1 and resolved this conversation.